src/odemis/gui/cont/acquisition.py

```diff
@@ -850,12 +829,16 @@ def _on_zstack(self):
         self._zlevels = {}
         for s in self._acquiStreams.value:
             if isinstance(s, FluoStream):
+                if not self._tab_data.main.focus:
+                    raise LookupError("Focuser was not found in the tab data")
                 levels = generate_zlevels(
                     self._tab_data.main.focus,
                     [self._tab_data.zMin.value, self._tab_data.zMax.value],
                     self._tab_data.zStep.value)
                 self._zlevels[s] = list(levels)
             elif isinstance(s, EMStream):
+                if not s.focuser:
+                    raise LookupError("The stream %s does not have SEM focuser object", s)
```

Comment:
I'm wondering, would it fail just because there is SEM stream? The SEM stream also has a focuser component. Or is that only in simulation? Technically, this is not essentially important, but i added these 2 if statements for readability and for understanding that the zstack is only for FM

Reply:
Anyway, this was removed, because the SEM stream is not included in the zlevels dictionary anymore